src/js/components/MenuList.js

```diff
@@ -0,0 +1,161 @@
+        $categoryName.innerHTML = CATEGORY.BLENDED;
+        break;
+      case "teavana":
+        $categoryName.innerHTML = CATEGORY.TEAVANA;
+        break;
+      case "dessert":
+        $categoryName.innerHTML = CATEGORY.DESSERT;
+        break;
+    }
+    menuCategory = categoryName;
+    $menuList.textContent = "";
+    $menuForm.reset();
+    loadMenu();
+    countMenu();
+  };
+
+  const countMenu = () => {
+    // 총 메뉴 갯수를 count하여 상단에 보여준다
+    const count = $all(".menu-list-item").length;
+    $menuCount.innerHTML = `총 ${count} 개`;
+  };
+
+  const drawMenu = (data) => {
+    const menuItem = createCustomElement(
+      "li",
+      "menu-list-item d-flex items-center py-2"
+    );
+    const menuItemName = createCustomElement("span", "w-100 pl-2 menu-name");
+    const menuItemSoldOutButton = createCustomButton(
+      ["mr-1", "menu-sold-out-button"],
+      (e) => soldOutMenu(e),
+      "품절"
+    );
+    // 메뉴의 수정 버튼을 눌러 메뉴 이름을 수정할 수 있다
+    const menuItemUpdateButton = createCustomButton(
+      ["mr-1", "menu-edit-button"],
+      (e) => updateMenu(e),
+      "수정"
+    );
+    // 메뉴 삭제 버튼을 이용하여 메뉴를 삭제할 수 있다
+    const menuItemDeleteButton = createCustomButton(
+      ["menu-remove-button"],
+      (e) => deleteMenu(e),
+      "삭제"
+    );
+    menuItem.id = data.id;
+    // 품절 상태인 경우를 보여줄 수 있게, 품절 버튼을 추가하고 sold-out class를 추가하여 상태를 변경한다
+    if (data.isSoldOut) menuItemName.classList.add("sold-out");
+    menuItemName.appendChild(document.createTextNode(data.name));
+    menuItem.appendChild(menuItemName);
+    menuItem.appendChild(menuItemSoldOutButton);
+    menuItem.appendChild(menuItemUpdateButton);
+    menuItem.appendChild(menuItemDeleteButton);
+    $menuList.appendChild(menuItem);
+  };
+
+  const saveMenu = () => {
+    localStorage.setItem(menuCategory, JSON.stringify(menuData));
+  };
+
+  const loadMenu = () => {
+    const savedMenuData = localStorage.getItem(menuCategory);
+    if (savedMenuData !== null) {
+      const parsedData = JSON.parse(savedMenuData);
+      parsedData.forEach(drawMenu);
+      menuData = parsedData;
+    } else menuData = [];
+  };
+
+  const createMenu = (e) => {
+    e.preventDefault();
+    // 사용자 입력값이 빈 값이라면 추가되지 않는다
+    if ($menuInput.value.trim() === "") return alert(MESSAGE.ALERT_CREATE);
+    const newMenu = {
+      id: `${Date.now()}`,
+      name: $menuInput.value,
+      isSoldOut: false,
+    };
+    drawMenu(newMenu);
+    menuData.push(newMenu);
+    saveMenu();
+    countMenu();
+    // 메뉴가 추가되고 나면, input은 빈 값으로 초기화한다
+    $menuForm.reset();
+  };
+
+  const updateMenu = ({ target }) => {
+    const $currentName = target.parentElement.querySelector(".menu-name");
+    // 메뉴 수정시 브라우저에서 제공하는 prompt 인터페이스를 활용한다
+    const updateName = prompt(MESSAGE.PROMPT_UPDATE, $currentName.innerHTML);
+    if (updateName) {
+      $currentName.innerHTML = updateName;
+      menuData.find((data) => {
+        if (data.id === target.parentElement.id) data.name = updateName;
+      });
```

Comment:
`find` 메서드를 독창적으로 사용하셨군요...!?
[find](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/find)메서드는
'배열 검색' 을 위해 사용되는 메서드입니다 ㅎㅎㅎ

```suggestion
const menu = menuData.find((menu /*여기서도 요소에 이름을 명확히 지어줄 수 있습니다*/) => menu.id === target.parentElement.id );

menu.name = updateName;
```

Reply:
아아 어떤 의미인지 이해했습니다. 검색된 반환값을 변수에 담는게 `find` 메서드를 사용하는 목적에 맞겠네요 !!